lib/commands/config.js

```diff
@@ -86,7 +86,7 @@ function defaultConfig() {
 
     // regex patterns
     config.convention.msgRegex = /^(?<type>\w+)(?:\((?<scope>[^()]+)\))?(?<breaking>!)?:\s*(?<description>.+)/i;
-    config.convention.msgMergeRegex = /^Merge branch +["'](?<branch>.+)['"]/i;
+    config.convention.msgMergeRegex = /(^Merge branch)|(^Merged in) +(["'])*(?<branch>.+)(['"])*/i;
```

Comment:
I'm not sure if it's a good idea to introduce app specific config into default config. You could set this regex within the project specific `git-conventional-commits.json` file

Reply:
My primary issue is the bad readability of the regex if we introduce more than one pattern by default, furthermore the regex gets more error prone.

Reply:
> I'm not sure if it's a good idea to introduce app specific config into default config. You could set this regex within the project specific `git-conventional-commits.json` file

My suggestion would be to put that as a commented out field in the config file and in the readme so it's more apparent that it can be changed outside of the code. 

As to app specific code, we already do that by defaulting to GitHub instead of Gitlab or something else. That said I 100% agree we shouldn't have app specific code just wasn't apparent to me that I was able to specify that in the config file.

Reply:
Damn it you were right it is not possible to override these regexpressions , however they should I think.

Reply:
Agreed 100%. It should all be configurable from that file. I'm going to close this PR in favor of a solution that makes it configurable from JSON. 

In addition, I introduced a bug in this PR but I'm not sure how I'm causing it. Tests aren't catching it but on one repo the changes work perfectly, on the other it gives me an error:
```
TypeError: Cannot read properties of undefined (reading 'replace')
    at escapeMarkdown (/Users/jdfalk/repos/github/jdfalk/git-conventional-commits/lib/changelogGenerator.js:226:17)
    at generateMarkdownCommit (/Users/jdfalk/repos/github/jdfalk/git-conventional-commits/lib/changelogGenerator.js:141:31)
    at /Users/jdfalk/repos/github/jdfalk/git-conventional-commits/lib/changelogGenerator.js:126:46
    at Array.forEach (<anonymous>)
    at /Users/jdfalk/repos/github/jdfalk/git-conventional-commits/lib/changelogGenerator.js:125:35
    at Array.forEach (<anonymous>)
    at generateMarkdownCommitTypeSection (/Users/jdfalk/repos/github/jdfalk/git-conventional-commits/lib/changelogGenerator.js:120:22)
    at /Users/jdfalk/repos/github/jdfalk/git-conventional-commits/lib/changelogGenerator.js:79:38
    at Array.forEach (<anonymous>)
    at module.exports.generateMarkdown (/Users/jdfalk/repos/github/jdfalk/git-conventional-commits/lib/changelogGenerator.js:78:25)
```